javav2/example_code/iotsitewise/src/main/java/com/example/iotsitewise/scenario/SitewiseActions.java

```diff
@@ -0,0 +1,479 @@
+
+    // snippet-start:[sitewise.java2_put_batch_property.main]
+    public CompletableFuture<BatchPutAssetPropertyValueResponse> sendDataToSiteWiseAsync(String assetId, String tempPropertyId, String humidityPropId) {
+        Map<String, Double> sampleData = generateSampleData();
+        long timestamp = Instant.now().toEpochMilli();
+
+        TimeInNanos time = TimeInNanos.builder()
+            .timeInSeconds(timestamp / 1000)
+            .offsetInNanos((int) ((timestamp % 1000) * 1000000))
+            .build();
+
+        BatchPutAssetPropertyValueRequest request = BatchPutAssetPropertyValueRequest.builder()
+            .entries(Arrays.asList(
+                PutAssetPropertyValueEntry.builder()
+                    .entryId("entry-3")
+                    .assetId(assetId)
+                    .propertyId(tempPropertyId)
+                    .propertyValues(Arrays.asList(
+                        AssetPropertyValue.builder()
+                            .value(Variant.builder()
+                                .doubleValue(sampleData.get("Temperature"))
+                                .build())
+                            .timestamp(time)
+                            .build()
+                    ))
+                    .build(),
+                PutAssetPropertyValueEntry.builder()
+                    .entryId("entry-4")
+                    .assetId(assetId)
+                    .propertyId(humidityPropId)
+                    .propertyValues(Arrays.asList(
+                        AssetPropertyValue.builder()
+                            .value(Variant.builder()
+                                .doubleValue(sampleData.get("Humidity"))
+                                .build())
+                            .timestamp(time)
+                            .build()
+                    ))
+                    .build()
+            ))
+            .build();
+
+        return getAsyncClient().batchPutAssetPropertyValue(request)
+            .whenComplete((response, exception) -> {
+                if (exception != null) {
+                    logger.error("An exception occurred: {}", exception.getCause().getMessage());
+                } else {
+                    logger.info("Data sent successfully.");
+                }
+            });
+    }
+    // snippet-end:[sitewise.java2_put_batch_property.main]
+
+    // snippet-start:[sitewise.java2_get_property.main]
+    // TODO -- fix this including Javadoc
+    public void getAssetPropValueAsync(String propName, String propId, String assetId) {
+        GetAssetPropertyValueRequest assetPropertyValueRequest = GetAssetPropertyValueRequest.builder()
+            .propertyId(propId)
+            .assetId(assetId)
+            .build();
+
+        CompletableFuture<GetAssetPropertyValueResponse> futureResponse = getAsyncClient().getAssetPropertyValue(assetPropertyValueRequest);
+        futureResponse.whenComplete((response, exception) -> {
```

Comment:
If you return the GetAssetPropertyValueResponse, why log the information here. Seems like the calling code should log it. 

I did notice when I reviewed it earlier that there is inconsistency in where information is logged from: sometimes it's logged in the Scenario and sometimes in the Action. We should be consistent and probably log the germain informtion in the scenario.

Reply:
Good comment - all logging should occur in the scenario class - not the Action class.